File: run_her2p.R
source("process_data.R")
source("methods.R")

args <- commandArgs(trailingOnly = TRUE)
st_id <- args[[1]] # e.g., "G2"
method <- args[[2]] # e.g., DOT, CARD, RCTD

sc_data <- load_sc_wu('HER2+')
st_data <- load_st_her2p(st_id)
annotations <- sc_data$metaa[, "celltype_major"]
locations <- st_data$meta[, c('x', 'y')]

if(method == "DOT")
{
  weights <- run_DOT(sc_data$counts, annotations, st_data$counts, locations,
                     max_spot_size = 200, normalize = FALSE) 
}else if(method == "CARD")
{
  weights <- run_CARD(sc_data$counts, annotations, st_data$counts, locations)  
}else if(method == "RCTD")
{
  weights <- run_RCTD(sc_data$counts, annotations, st_data$counts, locations)
}

File: run_her2p.py
from process_data import load_sc_wu, load_st_her2p
import sys

args = sys.argv
st_id = args[1] # e.g., "G2"
method = args[2] # e.g., TACCO, cell2location, ...

ref_adata = load_sc_wu("HER2+")
ref_annotation = "celltype_major"
srt_adata = load_st_her2p(st_id)

if method.lower() == "tacco":
    from methods import run_tacco
    weights = run_tacco(ref_adata, srt_adata, ref_annotation)
elif method.lower() in ["cell2location", "c2l"]:
    from methods import run_cell2location
    weights = run_cell2location(ref_adata, srt_adata, ref_annotation, 
                                epochs = 20000, normalize = False)